File: project_euler_problems/solved_problems/more_langs/PythonSolutions/problem_9/9.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-

"""
Project Euler Problem #9:

A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
a2 + b2 = c2

For example, 32 + 42 = 9 + 16 = 25 = 52.

There exists exactly one Pythagorean triplet for which a + b + c = 1000.
Find the product abc.
"""

for A in xrange(1, 1001):
	for B in xrange(A+1, 1001):
		for C in xrange(B+1, 1001):
			if A*A + B*B == C*C and A+B+C == 1000:
				print "A: %d\nB: %d\nC: %d" % (A, B, C)
				print "Sum: %d\nProduct: %d" % (A+B+C, A*B*C) #Answer: 31875000
				break

File: project_euler_problems/solved_problems/more_langs/C++Solutions/problem_9/9.cpp
#include <iostream>
using namespace std;

/* Project Euler #9:
 * A Pythagorean triplet is a set of three natural numbers, a < b < c,
 * for which,
 * a^2 + b^2 = c^2
 * For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
 *
 * There exists exactly one Pythagorean triplet for which a + b + c = 1000.
 * Find the product abc. */

int main()
{
    int product;
    for (int a = 0; a <= 1000; a++) {
        for (int b = (a + 1); b <= 1000; b++) {
            for (int c = (b + 1); c <= 1000; c++) {
                if (((a*a) + (b*b) == (c*c)) && (a + b + c) == 1000) {
                    product = a * b * c;
                }
            }
        }
    }
    // Answer: 31875000
    cout << "Product of abc in a + b + c = 1000: "
         << product << endl;
    return 0;
}